From f8e5619952fb25e648a91e7b616b7bcc91255299 Mon Sep 17 00:00:00 2001
From: Ben Dean-Kawamura <ben@pculture.org>
Date: Tue, 8 Nov 2011 15:35:29 -0500
Subject: [PATCH] Fix for killing dbus after running the unittests

I rewrote the script and split things up rather than having one huge line.
I'm not sure what the problem was before, but it was leaving me with an extra
dbus process each time I ran the unittests.  The new version works for me.
---
 tv/linux/test.sh |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/tv/linux/test.sh b/tv/linux/test.sh
index 1cce3e7..9e42e4a 100755
--- a/tv/linux/test.sh
+++ b/tv/linux/test.sh
@@ -35,4 +35,8 @@
 
 # this comes from http://www.estamos.de/blog/2009/05/08/running-syncevolution-as-cron-job/
 # env `dbus-launch` sh -c 'trap "kill $DBUS_SESSION_BUS_PID" EXIT; ./run.sh --unittest utiltest > /home/pcf/test_output.txt 1>&2' || true
-env `dbus-launch` bash -c "trap \"kill $DBUS_SESSION_BUS_PID\" EXIT; ./run.sh --unittest $1 $2 $3"
+
+
+eval `dbus-launch --sh-syntax`
+trap "kill $DBUS_SESSION_BUS_PID" EXIT
+./run.sh --unittest $@
-- 
1.7.9.1

